Documentation/planning/multi-sdk-band-support.md

```diff
@@ -0,0 +1,253 @@
+- The *outputs* of the runtime build combined with additional previously source-built artifacts (e.g. a matching band SDK, required tools and libraries) and source, to produce N SDK bands that meet source-build requirements.
+
+The runtime build requires a 1xx band SDK to build **and** redistributes functionality from tooling that ships in the 1xx band (e.g. it has dependencies on roslyn and roslyn-analyzers). Furthermore, most changes to a .NET major version occur when only one SDK band exists. Therefore it makes practical sense at this point to include the source of the 1xx SDK and the shared runtime in the same VMR branch, and to produce all artifacts related to 1xx, including the runtime, in its build. Newer bands will be VMRs that contain **only** the components that **differ** from the 1xx band. Only those components that differ. When building the newer band, the outputs of the 1xx build would be fed into the newer band build, in a similar manner as previously source built artifacts are today. If the source for a component does not exist in a given VMR, it is simply not built, and obviously no outputs would be produced for that component. Thus any downstream dependencies would be forced to use the versions provided by input source built artifacts, typically from the 1xx source build. If no such version exists, then the input would fail to be found or restored from online sources and be reported as a prebuilt.
+
+This approach fits naturally into the source-build methodology. Generally, additional SDK bands may be referred to as "subsetted VMRs", since use of a subsetted VMR is not restricted to SDKs only.
+
+### Conceptual splitting of 'previously source-built' artifact types
+
+Today, the term "previously source-built artifacts' refers to those artifacts that were built in a previous product build iteration. Typically N-1. For example, they would refer to the outputs of 8.0.1's build, when building 8.0.2. When building a subsetted VMR, two sets of input sets of source-built artifacts are required. While similar in some ways, they should be kept distinct for clarity. They are referred to as:
+
+- **Previously source-built (PSB)** - Artifacts from a previous servicing/preview iteration, or from a bootstrap build. These may be used when building the current VMR, but may not be bundled into any output.
+- **Current source-built (CSB)** - Artifacts just built, e.g. from a 1xx VMR build. These may be used **and** bundled into the build outputs, since logically they are from the same source-build iteration.
+
+### Supported input artifact combinations and validation thereof
+
+Not all toolsets can be used to build all VMRs. For instance, it is unlikely to be the case that the 1xx SDK will successfully build all 2xx+ bands. Components of newer SDKs frequently take dependencies on newer SDK features. However, the following statements can be made:
+
+- A newer SDK band (e.g. 2xx) may be buildable with an older band.
```

Comment:
Do we have some insights into the ratio of the bootstrapping / using older band scenario?

Reply:
I think it's going to be pretty rare. I do recall there having been cases where this happened (maybe .NET 6?), I think around NuGet or workload changes. But I can't recall any change in 7.0.Nxx that required a new SDK.